packages/s3-presigned-post/src/createPresignedPost.ts

```diff
@@ -66,13 +66,21 @@ export const createPresignedPost = async (
 
   // Prepare policies.
   const expiration = new Date(now.valueOf() + Expires * 1000);
-  const conditions: PolicyEntry[] = [
-    ...Conditions,
-    ...Object.entries(fields).map(([k, v]) => ({ [k]: v })),
-    Key.endsWith("${filename}")
-      ? ["starts-with", "$key", Key.substring(0, Key.lastIndexOf("${filename}"))]
-      : { key: Key },
-  ];
+
+  const coniditionsSet = new Set<string>();
+  Conditions.forEach((condition) => {
+    const stringifiedCondition = JSON.stringify(condition);
+    coniditionsSet.add(stringifiedCondition);
+  });
+
+  Object.entries(fields).forEach(([k, v]) => coniditionsSet.add(JSON.stringify({ [k]: v })));
+  if (Key.endsWith("${filename}")) {
+    coniditionsSet.add(JSON.stringify(["starts-with", "$key", Key.substring(0, Key.lastIndexOf("${filename}"))]));
+  } else {
+    coniditionsSet.add(JSON.stringify({ key: Key }));
+  }
+  const conditions = Array.from(coniditionsSet).map((item) => JSON.parse(item));
```

Comment:
this looks equivalent to
```js
const conditions = JSON.parse(Array.from(conditionsSet));
```

Reply:
nm

Reply:
`Array.from(conditionsSet)` returns an array of strings, this will cause `JSON.parse` to throw:

```js
JSON.parse(['{"key": 1}', '{"key": 2}', '{"key": 3}']) // not a valid json
```

 since it expects a single string that represents json object.